### DISCUSSION TITLE
Replicating objects to Google Cloud Storage

### DISCUSSION THREAD

**Discussion by fpetkovski** [minio/minio#20728] [Q&A] [answered]
I am trying to set up bucket replication from a minio bucket to a GCS bucket.

After enabling versioning on both ends, I get the following error message:
```
Remote service connection error (Remote service endpoint not available Health check timed out after 3 seconds)
```

MinIO successfully detected that both the source and remote buckets need versioning enabled. MinIO is also able to successfully authenticate with the remote bucket using the given credentials. Using invalid credentials leads to an authentication error.

Does MinIO assume that the remote bucket should also be hosted on a MinIO deployment? If so, it seems like an unnecessary limitation and very little work would be needed to support replicating to any object storage implementation.

**Comment by klauspost** [comment, marked as answer]
It is not possible to use bucket replication against non-MinIO targets as the S3 API does not provide the needed functionality.

The only option is to use client-side mirroring.

**Reply by fpetkovski** [reply]
I see, understood. Thank you for the fast response, I'll try setting up mirroring with `mc mirror` 👍

**Comment by marktheunissen** [comment]
@fpetkovski https://min.io/docs/minio/linux/administration/batch-framework-job-replicate.html

Batch framework can target GCS but it's not ongoing replication

**Reply by fpetkovski** [reply]
Thanks, I will look into this option as well.
